packages/eslint-config-airbnb/rules/react.js

```diff
@@ -524,9 +524,8 @@ module.exports = {
 
     // Enforce a specific function type for function components
     // https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/function-component-definition.md
-    // TODO: investigate if setting namedComponents to expression vs declaration is problematic
     'react/function-component-definition': ['error', {
-      namedComponents: 'function-expression',
+      namedComponents: 'function-declaration',
```

Comment:
can you file an upstream issue on eslint-plugin-react to allow this to be an array, so we can do `['function-declaration', 'function-expression']`?

Reply:
Opened a PR to make that change https://github.com/yannickcr/eslint-plugin-react/pull/3129